modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/AddToCart.js

```diff
@@ -71,6 +71,7 @@ function AddToCart({
 
 	const buttonDisabled = useMemo(() => {
 		if (
+			initialDisabled ||
```

Comment:
don't readd this beacuse this was the cause of the bug. For example now you may have a published SKU and with an Option change you may fetch an Unpublished sku
Add an || condition for published too.

Reply:
@andrea-ale-sbarra ,

Correct me if I am wrong, we would still need this property after adding `_published` because initialDisabled is the `_disabled` check from AddtoCartTag.java where it checks whether the button should be disabled following permission checks.